packages/server/src/api/rest/index.ts

```diff
@@ -0,0 +1,223 @@
+import { DataDocument, Serializer } from 'ts-japi';
+import { logError, RequestContext, stripAuxFields, zodValidate } from '../utils';
+
+/**
+ * JSON:API response
+ */
+export type Response = {
+    status: number;
+    body: unknown;
+    contentType?: string;
+};
+
+/**
+ * Prisma item
+ */
+export type Item = {
+    [key: string]: any;
+    id: any;
+};
+
+/**
+ * JSON:API serializers
+ */
+const serializers = new Map<string, Serializer>();
+
+/**
+ * Serialize Prisma items into JSON:API resources.
+ *
+ * @param items Item(s) to serialize
+ * @returns Serialize item(s)
+ */
+async function serializeItems(model: string, items: Item | Item[]): Promise<Partial<DataDocument<any>>> {
+    if (!serializers.has(model)) {
+        serializers.set(model, new Serializer(model));
+    }
+
+    stripAuxFields(items);
+
+    return JSON.parse(JSON.stringify(await serializers.get(model)!.serialize(items)));
+}
+
+/**
+ * Handles HTTP request
+ */
+export default async function handleRequest({
+    method,
+    path,
+    query,
+    requestBody,
+    prisma,
+    logger,
+    zodSchemas,
+}: RequestContext): Promise<Response> {
+    const parts = path.split('/').filter((p) => !!p);
+    const model = parts[0];
+    const resourceId = parts[1];
+
+    if ((parts.length !== 1 && parts.length !== 2) || !model) {
+        return {
+            status: 400,
+            body: { errors: [{ status: 400, code: 'invalid-path', title: 'Invalid request path' }] },
+        };
+    }
+
+    method = method.toUpperCase();
+
+    try {
+        if (method === 'GET') {
+            // Handling GET requests
+            if (!resourceId) {
+                const items = (await prisma[model].findMany({ where: query })) as Item[];
```

Comment:
`query` is actually the query parameters here, so not sure if passing it directly as `where` would work. The `Prisma` API handler used a `q` parameter to pass json-serialized prisma query. Do you it's good to do the same here?

Reply:
The `query` there was kept from the original Prisma API code as I had not worked on this part yet.

JSON:API is not prescriptive on the topic. They do recommend using `filter[field]` in the query-string for filtering (https://jsonapi.org/recommendations/#filtering). That would mean that, by default, we would only support exact equality. Other types of filtering (>, <, etc.) would be on us to define.

Generally, REST APIs do not seem to have any standard on how the query-string parameters should be used for filtering/sorting/pagination and I have seen a little bit of everything. Here is an example of how Feathers does it: https://feathersjs.com/api/databases/querying.html

Given that there is no strong standard, I am inclined to do it as you describe (accept a JSON-encoded field that can be passed directly to prisma `findMany` function) as that will make our life much easier (we automatically get all of Prisma's features on the topic). While having a JSON-encoded field in the query-string is not very REST, I think it's worth it given how much it simplifies the implementation.

What do you think?

Reply:
Yes, I completely agree with you. Using a json-encoded query parameter is both easy to implement and able to preserve Prisma's flexibility. In the future we can add support for JSON:API style filtering and use it as a shortcut and override.

Reply:
Hey @jawadst . I think the PR is in pretty good shape now. I'm going to merge it into the feature branch and continue making some tweaks there.

The main difference between the `Prisma` and `REST` is the usage of JSON:API standard. I'll probably merge these two implementations and add a switch to control the data protocol. In the future we can make the JSON:API flavor the default one.